File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio3/Java/Main.java
package Generecidad.Ejercicio3.Java;

public class Main {
    public static void main(String[] args) {
        System.out.println("--- Ejercicio 3: Clase Generica Catalogo ---");

        // b) 
        System.out.println("\n--- Probando Catalogo de Libros ---");
        Catalogo<Libro> catalogoLibros = new Catalogo<>(); 

        Libro libro1 = new Libro("Cien Años de Soledad", "Gabriel Garcia Marquez", "978-0307474474", 15.99);
        Libro libro2 = new Libro("Don Quijote de la Mancha", "Miguel de Cervantes", "978-8424118029", 12.50);
        Libro libro3 = new Libro("1984", "George Orwell", "978-0451524935", 9.50);

        catalogoLibros.agregarElemento(libro1);
        catalogoLibros.agregarElemento(libro2);
        catalogoLibros.agregarElemento(libro3);

        catalogoLibros.mostrarCatalogo();

        Libro libroEncontrado1 = catalogoLibros.buscarElemento("cien años");
        if (libroEncontrado1 != null) {
            System.out.println("Encontrado: " + libroEncontrado1);
        } else {
            System.out.println("Libro no encontrado.");
        }

        Libro libroEncontrado2 = catalogoLibros.buscarElemento("84241");
        if (libroEncontrado2 != null) {
            System.out.println("Encontrado: " + libroEncontrado2);
        } else {
            System.out.println("Libro no encontrado.");
        }

        Libro libroNoEncontrado = catalogoLibros.buscarElemento("El Principito");
        if (libroNoEncontrado != null) {
            System.out.println("Encontrado: " + libroNoEncontrado);
        } else {
            System.out.println("Libro 'El Principito' no encontrado.");


        }
        // c) 
        System.out.println("\n--- Probando Catalogo de Productos ---");
        Catalogo<Producto> catalogoProductos = new Catalogo<>(); 

        Producto producto1 = new Producto("Monitor HP 24'", "MON-HP-001", 250.00);
        Producto producto2 = new Producto("Teclado Mecanico RGB", "TEC-GAMER-005", 85.99);
        Producto producto3 = new Producto("Mouse Optico Inalámbrico", "MOU-WIRE-010", 25.00);

        catalogoProductos.agregarElemento(producto1);
        catalogoProductos.agregarElemento(producto2);
        catalogoProductos.agregarElemento(producto3);

        catalogoProductos.mostrarCatalogo();

        Producto productoEncontrado1 = catalogoProductos.buscarElemento("monitor");
        if (productoEncontrado1 != null) {
            System.out.println("Encontrado: " + productoEncontrado1);
        } else {
            System.out.println("Producto no encontrado.");
        }

        Producto productoEncontrado2 = catalogoProductos.buscarElemento("GAMER-005");
        if (productoEncontrado2 != null) {
            System.out.println("Encontrado: " + productoEncontrado2);
        } else {
            System.out.println("Producto no encontrado.");
        }

        Producto productoNoEncontrado = catalogoProductos.buscarElemento("Webcam");
        if (productoNoEncontrado != null) {
            System.out.println("Encontrado: " + productoNoEncontrado);
        } else {
            System.out.println("Producto 'Webcam' no encontrado.");
        }
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio3/Python/main.py
from Catalogo import Catalogo
from Libro import Libro
from Producto import Producto

if __name__ == "__main__":
    print("--- Ejercicio 3: Clase Generica Catalogo ---")

    # b) 
    print("\n--- Probando Catalogo de Libros ---")
    catalogo_libros = Catalogo[Libro]() 

    libro1 = Libro("Cien Años de Soledad", "Gabriel Garcia Marquez", "978-0307474474", 15.99)
    libro2 = Libro("Don Quijote de la Mancha", "Miguel de Cervantes", "978-8424118029", 12.50)
    libro3 = Libro("1984", "George Orwell", "978-0451524935", 9.50)

    catalogo_libros.agregar_elemento(libro1)
    catalogo_libros.agregar_elemento(libro2)
    catalogo_libros.agregar_elemento(libro3)

    catalogo_libros.mostrar_catalogo()

    libro_encontrado_1 = catalogo_libros.buscar_elemento("cien años")
    if libro_encontrado_1:
        print(f"Encontrado: {libro_encontrado_1}")
    else:
        print("Libro no encontrado.")

    libro_encontrado_2 = catalogo_libros.buscar_elemento("84241")
    if libro_encontrado_2:
        print(f"Encontrado: {libro_encontrado_2}")
    else:
        print("Libro no encontrado.")

    libro_no_encontrado = catalogo_libros.buscar_elemento("El Principito")
    if libro_no_encontrado:
        print(f"Encontrado: {libro_no_encontrado}")
    else:
        print("Libro 'El Principito' no encontrado.")


    # c) 
    print("\n--- Probando Catalogo de Productos ---")
    catalogo_productos = Catalogo[Producto]() 

    producto1 = Producto("Monitor HP 24'", "MON-HP-001", 250.00)
    producto2 = Producto("Teclado Mecanico RGB", "TEC-GAMER-005", 85.99)
    producto3 = Producto("Mouse Optico Inalambrico", "MOU-WIRE-010", 25.00)

    catalogo_productos.agregar_elemento(producto1)
    catalogo_productos.agregar_elemento(producto2)
    catalogo_productos.agregar_elemento(producto3)

    catalogo_productos.mostrar_catalogo()

    producto_encontrado_1 = catalogo_productos.buscar_elemento("monitor")
    if producto_encontrado_1:
        print(f"Encontrado: {producto_encontrado_1}")
    else:
        print("Producto no encontrado.")

    producto_encontrado_2 = catalogo_productos.buscar_elemento("GAMER-005")
    if producto_encontrado_2:
        print(f"Encontrado: {producto_encontrado_2}")
    else:
        print("Producto no encontrado.")

    producto_no_encontrado = catalogo_productos.buscar_elemento("Webcam")
    if producto_no_encontrado:
        print(f"Encontrado: {producto_no_encontrado}")
    else:
        print("Producto 'Webcam' no encontrado.")
